src/data/template/Microsoft.Authorization/policyAssignments.template.jq

```diff
@@ -20,4 +20,8 @@
     ],
     "outputs": {}
 } | .resources[] |= if .identity==null then del(.identity) elif .identity.UserAssignedIdentities==null then del(.identity.UserAssignedIdentities) else . end | 
- .resources[] |= if .identity.IdentityType !=null then .identity["Type"] = .identity.IdentityType | del(.identity.IdentityType) else . end
+ .resources[] |= if .identity.IdentityType !=null then .identity["Type"] = .identity.IdentityType | del(.identity.IdentityType) else . end |
+ .resources[] |= if .identity==null then del(.identity) elif .identity.UserAssignedIdentities==null then del(.identity.UserAssignedIdentities) else . end |
+ .resources[] |= if .identity.UserAssignedIdentities != null then del( .identity.UserAssignedIdentities[].PrincipalId?, .identity.UserAssignedIdentities[].ClientId?, .identity.ClientId?, .identity.PrincipalId?) else . end |
+ .resources[] |= if .identity.PrincipalId == null then del(.identity.PrincipalId) else . end |
+ .resources[] |= if .identity.TenantId == null then del(.identity.TenantId) else . end
```

Comment:
A couple of things: 
- Line 22 have been duplicated on row 24
- `.identity.ClientId` should be `.identity.TenantId` on row 25
- No need to explicitly check for `.identity.PrincipalId` and `.identity.TenantId` since we only want to delete those properties in the UMI case. If case of a system managed identity, they will be populated, if identity is null, we will delete the property. 
```suggestion
.resources[] |= if .identity.IdentityType !=null then .identity["Type"] = .identity.IdentityType | del(.identity.IdentityType) else . end |
.resources[] |= if .identity.UserAssignedIdentities != null then del(.identity.UserAssignedIdentities[].PrincipalId, .identity.UserAssignedIdentities[].ClientId, .identity.TenantId, .identity.PrincipalId) else . end
```
I have tried push/pull of UMI/SMI DINE policies with the above changes, and seems to be working as expected.

Reply:
Agreed! Thanks for reviewing - I was too lazy and not up to par there but learned some jq in the process.